modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/NodAndShuffleProgress.scala

```diff
@@ -135,38 +136,39 @@ sealed trait NodAndShuffleProgress {
   // during your components lifecycle ensures that React will update your
   // component rather than unmounting and remounting it."
   @Lenses
-  protected case class State(progressConnect: ReactConnectProxy[Option[Progress]])
+  protected case class State(progressConnect: ReactConnectProxy[Option[NSObservationProgress]])
 
   implicit val propsReuse: Reusability[Props]
   implicit val stateReuse: Reusability[State] = Reusability.always
 
   protected def sections(nsStatus: NodAndShuffleStatus): List[DividedProgress.Label]
 
   protected def quantitiesFromNodMillis(
-    nsStatus:  NodAndShuffleStatus,
-    nodMillis: DividedProgress.Quantity
-  ): (DividedProgress.Quantity, Option[DividedProgress.Quantity]) // (sectionTotal, currentValue)
+    cycleIndex: Int,
+    nodIndex:   Int,
+    nodMillis:  DividedProgress.Quantity
+  ): (DividedProgress.Quantity, DividedProgress.Quantity) // (sectionTotal, currentValue)
 
   protected[steps] val component = ScalaComponent
     .builder[Props]("NodAndShuffleProgress")
     .initialStateFromProps(p => State(p.connect))
     .render_PS { (p, s) =>
       s.progressConnect { proxy =>
-        val (totalMillis, remainingMillis) =
+        val (totalMillis, remainingMillis, cycleIndex, nodIndex) =
           proxy().foldMap(p =>
-            (p.total.toMilliseconds.toInt, p.remaining.toMilliseconds.toInt))
+            (p.total.toMilliseconds.toInt, p.remaining.toMilliseconds.toInt, p.sub.cycle.toInt, p.sub.stageIndex))
         val elapsedMillis = totalMillis - max(0, remainingMillis)
 
         p.summary.nsStatus.map[VdomElement] { nsStatus =>
           val isInError = !p.summary.isNSRunning && p.summary.isNSInError
           val nodMillis = nsStatus.nodExposureTime.toMilliseconds.toInt
-          val (sectionTotal, currentValue) = quantitiesFromNodMillis(nsStatus, nodMillis)
+          val (sectionTotal, currentValue) = quantitiesFromNodMillis(cycleIndex, nodIndex, nodMillis)
 
           SmoothDividedProgressBar(
             sections = sections(nsStatus),
             sectionTotal = sectionTotal,
-            value = currentValue.map(_ + elapsedMillis).getOrElse(0),
-            maxValue = currentValue.map(_ + nodMillis).getOrElse(0),
+            value = nsStatus.state.map(_ => currentValue + elapsedMillis).orEmpty, // Only advance smooth bar if actually started

```

Comment:
`foldMap`

Reply:
Right, keep forgetting about that! Done.